src/main/groovy/io/codearte/gradle/nexus/infra/SimplifiedHttpJsonRestClient.groovy

```diff
@@ -51,8 +53,18 @@ class SimplifiedHttpJsonRestClient {
         Map params = createAndInitializeCallParametersMap()
         params.body = content
         log.debug("POST request content: $content")
-        //TODO: Add better error handling (e.g. display error message received from server, not only 500 + not fail on 404 in 'text/html')
-        HttpResponseDecorator response = (HttpResponseDecorator)restClient.post(params)
-        log.warn("POST response data: ${response.data}")
+		try {
+			//TODO: Add better error handling (e.g. display error message received from server, not only 500 + not fail on 404 in 'text/html')
+			HttpResponseDecorator response = (HttpResponseDecorator)restClient.post(params)
+			log.warn("POST response data: ${response.data}")
+		} catch(groovyx.net.http.HttpResponseException e) {
+		    //Apache' HttpResponseException ONLY puts the 2nd param in the e.getMessage which will be printed so
+		    //put all information there (status code, error str, body of response in case they put more error information there)
+		
+			HttpResponseDecorator resp = e.getResponse();
+			String message = "${resp.statusLine.statusCode}:${resp.statusLine.reasonPhrase}  body=${resp.data}"
+		    log.error("POST response failed.  ${message}")
+			throw new HttpResponseException(e.getStatusCode(), message)
+		}
```

Comment:
Please reformat this code fragment in general. It seems your are using tabs instead of spaces.